### DISCUSSION TITLE
virtual service visibility in multi cluster , multi primary model

### DISCUSSION THREAD

**Discussion by nadeemhussain** [istio/istio#48768] [Q&A] [answered]
Hi team , we have multiple k8s cluster each has there own istiod on it , with flat network we installed secrets on each cluster for other clusters info to have a the mesh across the clusters.

this works great and app in cluster A can access app in cluster B , but the virtual service in Cluster A is not visible in other cluster ,  is there a way to make it visible across clusters or these need to be created explicitly

**Comment by linsun** [comment, marked as answer]
I don't think there is a simple way other than building some automation around it - for example, using your CI/CD system to also deploy the VS to other clusters where the VS is needed.

**Reply by nadeemhussain** [reply]
that sounds about right , btw is there any specifc reason why istio dont support this,
 as istiod  of cluster A fetches svc/endpoints/nodes and other things from api server of cluster B  and distributes to  its clusters envoy proxies , why can't it fetch vs/destinationrule too , so this would have been supported natively at istio level then hacking our way around with differnt tools to achieve this , i might be missing some reason why this design decision was made .

**Reply by tsl409** [reply]
Agreed. This is so counter-intuitive. Should at least mentioned somewhere in documentation what is propagated and what is not or maybe I missed it? 🤔

**Reply by gjbravi** [reply]
For people who may land on this discussion, my 2 cents on this: [Karmada resource propagation](https://karmada.io/docs/userguide/scheduling/resource-propagating/) is useful in scenarios where resources like a VirtualService or DestinationRule need to be propagated across clusters.

As, apparently, by design, istio doesn't do that natively.
